types/types.go

```diff
@@ -23,67 +24,158 @@ import (
 	"github.com/prometheus/common/model"
 )
 
+type state uint8
+
+const (
+	Unprocessed state = iota
```

Comment:
Yeah I'm not sure what name to provide. I don't feel strongly one way or the other.